packages/flutter/lib/src/services/text_input.dart

```diff
@@ -633,6 +633,9 @@ abstract class TextInputClient {
 
   /// Updates the floating cursor position and state.
   void updateFloatingCursor(RawFloatingCursorPoint point);
+
+  /// Platform notified framework of closed connection due to user activity.
```

Comment:
Can we add a little more documentation here: e.g. when this notification is received the connection should no longer be used or something like that.

Also, what's meant by "due to user activity"? This can be sent whenever the platform wants to close the connection - even if that is not in response to user activity, no?

Reply:
How will we differentiate between the platform sending this because it's about to change the style of the keyboard, and the platform doing this because the user has asked it to dismiss the keyboard permanently?

Reply:
The platform messages sent by the engines (Android, Web etc) are called TextInputClient.methodName. Such as TextInputClient.performAction or TextInputClient.updateEditingState

The framework send methods are TextInput.methodName such as TextInput.clearClient, TextInput.hide, TextInput.show

Thanks for bringing this up!

Reply:
Added more comments.